src/transformers/models/detr/feature_extraction_detr.py

```diff
@@ -547,6 +548,12 @@ def __call__(
             if annotations is not None:
                 annotations = [annotations]
 
+        # Create deep copies to avoid editing inputs in place
+        images = [copy.deepcopy(image) for image in images]
+
+        if annotations is not None:
+            annotations = [copy.deepcopy(annotation) for annotation in annotations]
```

Comment:
You don't need this expensive `deepcopy` operation. The reason the inputs are modified is because, unlike in other feature extractors, the same list is used in the call and the objects at each index are replaced. I don't think the later processing logic modifies the images or annotations in place, so you can just do: 

```suggestion
        # Create copies to avoid editing inputs in place
        images = [image for image in images]           
        annotations = [annotation for annotation in annotations]
```